lightwood/analysis/nc/icp.py

```diff
@@ -194,6 +195,7 @@ def predict(self, x, significance=None):
         # TODO: if x == self.last_x ...
         n_test_objects = x.shape[0]
         p = np.zeros((n_test_objects, self.classes.size))
+        random_uniform_dist = np.random.uniform(0, 1, 1)
         for i, c in enumerate(self.classes):
```

Comment:
What is the use of the uniform dist here?

Reply:
See section 5, eq. 13 [here](http://alrw.net/articles/02.pdf) for the reason. 

We actually need to sample a different tie-breaker value for every `p[j, i]`, so the optimization here introduces a bug, please revert it so that `np.random.uniform` gets called for each `cal_score`.

Reply:
Additionally, please make that citation as a comment or within the docstring because for future use, we may want to refer to the research rationale here.